test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs

```diff
@@ -101,5 +103,31 @@ public async Task ApplySchema_UsingDiffScript_Succeeds()
             await _sqlSchemaManager.ApplySchema("connectionString", new Uri("https://localhost/"), new MutuallyExclusiveType { Latest = false, Version = 2, Next = false });
             await _schemaManagerDataStore.Received().ExecuteScriptAndCompleteSchemaVersionAsync(Arg.Is("script"), Arg.Is(2), Arg.Any<CancellationToken>());
         }
+
+        [Fact]
+        public async Task ApplySchema_OnDependencyThrowSchemaManagerException_ThrowsSchemaManagerException()
+        {
+            // Set a zero retry sleep duration to expedite fail-case unit test.
+            FieldInfo retrySleepDurationField = typeof(SqlSchemaManager).GetField("_retrySleepDuration", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
>_retrySleepDuration [](http://example.com/codeflow?start=83&length=19)

Wouldn't it be better to make this part of configuration instead doing reflection magic?
`SqlServerSchemaOptions` sounds like good place to hold it.

Reply:
I considered that, but with this pattern I am avoiding having a requirement that is specific to unit tests leak into the public API.

Reply:
@Ivanidzo4ka, Please re-open if you feel strongly and I will update it to use SqlServerSchemaOptions.